packages/contractkit/src/wrappers/Exchange.ts

```diff
@@ -1,18 +1,125 @@
 import BigNumber from 'bignumber.js'
 import { Exchange } from '../generated/types/Exchange'
-import { BaseWrapper, proxyCall, proxySend, toBigNumber } from './BaseWrapper'
+import {
+  BaseWrapper,
+  CeloTransactionObject,
+  NumberLike,
+  parseNumber,
+  proxyCall,
+  proxySend,
+  toBigNumber,
+  tupleParser,
+} from './BaseWrapper'
 
+/**
+ * Contract that allows to exchange StableToken for GoldToken and vice versa
+ * using a Constant Product Market Maker Model
+ */
 export class ExchangeWrapper extends BaseWrapper<Exchange> {
-  getBuyTokenAmount = proxyCall(this.contract.methods.getBuyTokenAmount, undefined, toBigNumber)
+  /**
+   * @dev Returns the amount of buyToken a user would get for sellAmount of sellToken
+   * @param sellAmount The amount of sellToken the user is selling to the exchange
+   * @param sellGold `true` if gold is the sell token
+   * @return The corresponding buyToken amount.
+   */
+  getBuyTokenAmount: (sellAmount: NumberLike, sellGold: boolean) => Promise<BigNumber> = proxyCall(
+    this.contract.methods.getBuyTokenAmount,
+    tupleParser(parseNumber, (x: boolean) => x),
```

Comment:
Did you mean to use the new noopParser here? Ditto in a few spots below.

Reply:
originally, but didn't work. I had to specify the types to make it a valid call